### PR TITLE
feat: ability to set initial format for webcomponent

### LINKED ISSUE #16: WebComponent - Defaults to Still Image. No Autoplay. Can't Dismiss stats
**Describe the bug**
When I import the minified file (media-stream-player.min.js) and add the webcomponent with the host hard-coded there are a few issues I encounter:
1) There appears to be no autoplay for the webcomponent. I must hit the play button.
2) When I hit play, it defaults to "still image" even though settings shows H264 over RTP selected for the format.
3) When I press the stats button, the overlay appears. Pressing the button again does not hide the stats. Requires a full page refresh to hide the stats.

**To Reproduce**
```html
<!DOCTYPE html>
<html>
<head>
    <title>Example video streaming from Axis camera</title>
</head>

<body>
<div style="display: flex; width: 100%">
    <media-stream-player hostname="<my ip>" style="height:100%;width:100%;"/>
</div>
<script src="media-stream-player.min.js"></script>
</body>
</html>
```

1. Set up: Above code.
2. Do this: Press play. Press configuration and stats button. Notice "Type" is "still image"
3. Change format to still image, then back to H.264 over RTP. Now playing video.

**Screenshots**

**Environment (please complete the following information):**
 - OS: MacOS 10.15.4
 - Browser: Chrome and Brave
 - Version: Compiled from master branch today.

### PR DESCRIPTION
**Pull request by lekoaf**
- Ability to set initial format for webcomponent
- H264 is registered as H264, not MJPEG
- Format dropdown now initializes with correct format

Relates to #51.
Fixes #16.

### CODE REVIEW

File: examples/web-component/index.html [hunk 1]
```diff
@@ -5,10 +5,21 @@
...
+      html {
+        height: 98vh;
+      }
+      body {
+        height: 98vh;
+      }
+      #root {
+        height: 98vh;
+      }
+    </style>
```
Comment (lekoaf) [inline examples/web-component/index.html:18]: I'm working on a solution for this "problem".

Comment (lekoaf) [reply examples/web-component/index.html]: I think this is the best I can do right now.

File: lib/Player.tsx [hunk 2]
```diff
@@ -167,7 +165,7 @@ export const Player = forwardRef<PlayerNativeElement, PlayerProps>(
       setWaiting(false)
     }, [])
 
-    const onFormat = (format: Format) => {
+    const onFormat = useCallback((format: Format | undefined) => {
```
Comment (steabert) [inline lib/Player.tsx:184]: Looks like this can just be inside the `useEffect`, then you don't need a separate `useCallback`?.

**Comment by coricidin12** [comment]
Has this been fixed? I'm trying to use the web component to autoplay the H264 stream but it keeps defaulting to still image. Will it require a new media-stream-player.min.js file?

Also will there be a way to set initial values for compression and resolution? Or alternatively javascript functions that could do that rather than requiring user interaction?

**Comment by lekoaf** [comment]
Almost. Just need @steabert to review, merge and release it first.

**Comment by lekoaf** [comment]
> Also will there be a way to set initial values for compression and resolution? Or alternatively javascript functions that could do that rather than requiring user interaction?

Nope, but we can fix that as well.

**Comment by lekoaf** [comment]
Updated the readme with a bit more elaborate instructions on how to use the web-component. It seems to be popular.

**Comment by coricidin12** [comment]
thanks, but doesn't the media-stream-player.min.js file need to be updated to make this work for the web component? When I download this commit, that file only contains `../../dist/media-stream-player.min.js`

I tried running it with the python server suggested in the readme, and get an error in the console when it tries to load that file. 

I don't know much about react, but don't the .tsx files updated in this commit only relate to the react component use?

**Comment by lekoaf** [comment]
Yeah you need to build it manually if you want these changes before they are merged and released.

**Review by barnomasoz — APPROVED** [review]
Looks okay

**Comment by steabert** [comment]
Fixes #31

**Comment by lekoaf** [comment]
@coricidin12, we just did a new release. So you can download the latest minfied file now.
https://github.com/AxisCommunications/media-stream-player-js/releases/latest

**Comment by coricidin12** [comment]
Works great, thank you!

**Pull request merged**